### PR TITLE
Ocs 2199

### PR DESCRIPTION
**Pull request by gpotts**
Initial add of Blocked stream reads.  This can adapt any stream and do block reads

### CODE REVIEW

File: src/base/ossimBlockStreamBuffer.cpp [hunk 1]
```diff
@@ -0,0 +1,270 @@
...
+                         std::ios_base::seekdir dir,
+                         std::ios_base::openmode mode)
+{
+   // make sure we are in synch with current pos
+   // gptr can be updated by other means
+   syncCurrentPosition();
+   pos_type result = pos_type(off_type(-1));
+   if(m_adaptStream)
+   {
+      // result = m_adaptStream->rdbuf()->pubseekoff(offset, dir, mode);
```
Comment (Spectrality) [inline src/base/ossimBlockStreamBuffer.cpp]: will this commented out code be used at some point?

File: src/base/ossimBlockStreamBuffer.cpp [hunk 2]
```diff
@@ -0,0 +1,270 @@
...
+:  m_adaptStream(adaptStream),
+   m_currentPosValue(0)
+{
+   m_blockInfo.m_blockSize = blockSize;
+   if(m_blockInfo.m_blockSize)
+   {
+      m_blockBuffer.resize(m_blockInfo.m_blockSize);
+      m_blockInfo.m_blockBufferPtr = &m_blockBuffer.front();
+
+//      setbuf(m_blockInfo.m_blockBufferPtr, m_blockInfo.m_blockSize);
```
Comment (Spectrality) [inline src/base/ossimBlockStreamBuffer.cpp]: does this commented out code need to stay here?

File: src/base/ossimBlockStreamBuffer.cpp [hunk 3]
```diff
@@ -0,0 +1,270 @@
...
+         }
+         case std::ios_base::cur:
+         {
+            pos += offset;
+            result = seekpos(pos, std::ios_base::in);
+           break;
+         }
+         case std::ios_base::end:
+         {
+            //if(!m_adaptStream->good()) m_adaptStream->clear();
```
Comment (Spectrality) [inline src/base/ossimBlockStreamBuffer.cpp]: this commented code will be used if the offset is known? or is the comment below unrelated to this commented out code?

Comment (gpotts) [reply src/base/ossimBlockStreamBuffer.cpp]: Removed the code and will do a new pull

File: src/base/ossimBlockStreamBuffer.cpp (hunk 1 above)
Comment (gpotts) [reply src/base/ossimBlockStreamBuffer.cpp]: removed the code and will do a pull

**Pull request merged**